scipy/stats/_distn_infrastructure.py

```diff
@@ -1421,13 +1421,25 @@ def support(self, *args, **kwargs):
             scale parameter, Default is 1.
         Returns
         -------
-        a, b : float
+        a, b : array_like
             end-points of the distribution's support.
 
         """
         args, loc, scale = self._parse_args(*args, **kwargs)
+        arrs = np.broadcast_arrays(*args, loc, scale)
+        args, loc, scale = arrs[:-2], arrs[-2], arrs[-1]
+        cond = self._argcheck(*args) & (scale > 0)
         _a, _b = self._get_support(*args)
-        return _a * scale + loc, _b * scale + loc
+        if cond.all():
+            return _a * scale + loc, _b * scale + loc
+        elif cond.ndim == 0:
+            return self.badvalue, self.badvalue
+        # promote bounds to at least float to fill in the badvalue
+        _a, _b = 1.*_a, 1.*_b
```

Comment:
I haven't looked into how `place` works and exactly what kind of NaN `self.badvalue` can be, but if `self.badvalue` is always a `np.float64`, is it reasonable to just promote to that?

Reply:
> if `self.badvalue` is always a `np.float64`, is it reasonable to just promote to that?

IIUC, we can also promote to `np.float64` but it won't make any difference as `place` will safely coerce the datatype of `self.badvalue` to match that of the result in which we are placing it.

> Do we support higher data types than that in stats?

AFAIK, the highest supported datatype is `np.float64`.

Reply:
I'm getting confused. What is the advantage of multiplying by `1.` instead of explicitly promoting to `np.float64`?

Reply:
Sometimes the bounds returned by the distribution are python `float`s/`int`s (even for array inputs). So, we need to do something like `np.asarray(_a).astype('d')` to be able to convert the bounds to `np.float64`. I thought it would be easier to coerce any bounds to float by multiplying by `1.`. I am also fine with the former approach. What do you say? Should I change it to that?

Reply:
OK. Sometimes I agree that it is cleaner to do things like this. In this case, though, I think it is more readable (more obvious precisely what the effect is) and more efficient (no multiplying by 1.0) to use `np.asarray(_a, dtype='d')` as long as we're not concerned with supporting `float128`. It will also still fit in one line, so I think I'd prefer that.

Reply:
> OK. Sometimes I agree that it is cleaner to do things like this. In this case, though, I think it is more readable (more obvious precisely what the effect is) and more efficient (no multiplying by 1.0) to use np.asarray(_a, dtype='d') as long as we're not concerned with supporting float128. It will also still fit in one line, so I think I'd prefer that.

Done. Thanks for the reply!

Reply:
I'll take another look soon.